crates/codegen/parser/runtime/src/cursor.rs

```diff
@@ -6,68 +6,54 @@ use crate::cst::{NamedNode, Node, RuleNode};
 use crate::kinds::{FieldName, RuleKind, TokenKind};
 use crate::text_index::{TextIndex, TextRange};
 
-/// A [`PathNode`] that points to a [`RuleNode`].
+/// A node in the ancestor path of a [`Cursor`].
 #[derive(Clone, Debug, PartialEq, Eq)]
-struct PathRuleNode {
+struct PathAncestor {
```

Comment:
Love the change!

But given that it is a major rewrite, I wonder if it is a good opportunity to write some tests .. simply covering the public calls/APIs.

Reply:
I thought about that before the rewrite, but I figured that this is so fundamental an API that the other layered subsystems are already effectively testing it. Happy to revisit later, for completeness, but don't think it is a practical requirement right now.